parsec/serde/fields.py

```diff
@@ -105,16 +104,16 @@ def enum_field_factory(enum: Type[Enum]) -> Type[BaseEnumField]:
     return type(f"{enum.__name__}Field", (BaseEnumField,), {"ENUM": enum})
 
 
-def bytes_based_field_factory(value_type: Type) -> Type[Field]:
+def bytes_based_field_factory(value_type: Any) -> Type[Field]:
     assert isinstance(value_type, type)
 
-    def _serialize(self, value, attr, obj):
+    def _serialize(self: Any, value: Any, attr: Any, obj: Any) -> bytes | None:
         if value is None:
             return None
```

Comment:
> :raise ValidationError: In case of formatting or validation failure.

Do you think this should raise a `ValidationError` or return `None` is fine too?

Reply:
Out of scope